microcosm_postgres/operations.py

```diff
@@ -65,13 +66,34 @@ def drop_alembic_table(graph):
         return True
 
 
+# Cached database metadata instance
+_metadata = None
+
+
 def recreate_all(graph):
     """
-    Drop and add back all database tables.
+    Drop and add back all database tables, or reset all data associated with a database.
+    Intended mainly for testing, where a test database may either need to be re-initialized
+    or cleared out between tests
 
     """
-    drop_all(graph)
-    create_all(graph)
+
+    global _metadata
+
+    if _metadata is None:
+        # First-run, the test database/metadata needs to be initialized
+        drop_all(graph)
+        create_all(graph)
+
+        _metadata = MetaData(bind=graph.postgres, reflect=True)
+        return
+
+    # Otherwise, truncate all existing tables
```

Comment:
Seems like you could reset sequences if you were sufficiently motivated.

Reply:
The motivation would mainly be to reduce the amount of unit tests I'd have to fix after this change rolls out. There probably aren't *that* many that do direct comparisons, but probably enough to be annoying.

Reply:
I tried out a variant of https://stackoverflow.com/questions/23038422/reset-id-count-after-table-delete. It works, but is substantially slower (6s => 3s alone in these unit tests).